out/main.go

```diff
@@ -4,19 +4,34 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"io/ioutil"
 	"log"
 	"os"
+	"path"
 	"strconv"
+	"strings"
 
 	"github.com/arbourd/concourse-slack-alert-resource/concourse"
 	"github.com/arbourd/concourse-slack-alert-resource/slack"
 )
 
 func buildMessage(alert Alert, m concourse.BuildMetadata) *slack.Message {
 	fallback := fmt.Sprintf("%s -- %s", fmt.Sprintf("%s: %s/%s/%s", alert.Message, m.PipelineName, m.JobName, m.BuildName), m.URL)
+	PutBasePath := os.Args[1]
```

Comment:
@arbourd let me know what you think of this, I tested it (by pushing it to my own private dockerhub repo and trying it locally)

Reply:
I have one question tho regarding go, is there a best practice when declaring vars in go to protect from a bad return of a function to avoid a panic? So for example in ruby when you `temp = ARGV[1] || "YO"` and there are no arguments to the scrip (i.e. ARGV[1]=nil) that works out. Is there something similar in go to protect from a panic if `os.Args[1]` does not exist? 

This is not be needed in this scenario since concourse ALWAYS passes an argument while executing ./out but I'm just curious about what a best practice would be in general #asking4aNoobFrend

Reply:
That's a great question. In this case, args[1] should never be empty and I'd leave it there because it's a big failure.

For a Go-y check of an array though, we'd probably check the length with `len()` before trying to access it.